oci/layout/oci_transport.go

```diff
@@ -176,16 +179,54 @@ func (ref ociReference) PolicyConfigurationNamespaces() []string {
 // NOTE: If any kind of signature verification should happen, build an UnparsedImage from the value returned by NewImageSource,
 // verify that UnparsedImage, and convert it into a real Image via image.FromUnparsedImage.
 func (ref ociReference) NewImage(ctx *types.SystemContext) (types.Image, error) {
-	src := newImageSource(ref)
+	d, err := ref.getManifestDescriptor()
+	if err != nil {
+		return nil, err
+	}
+	src := newImageSource(ref, d)
 	return image.FromSource(src)
 }
 
+func (ref ociReference) getManifestDescriptor() (imgspecv1.ManifestDescriptor, error) {
+	indexJSON, err := os.Open(ref.indexPath())
+	if err != nil {
+		return imgspecv1.ManifestDescriptor{}, err
+	}
+	defer indexJSON.Close()
+	index := imgspecv1.ImageIndex{}
+	if err := json.NewDecoder(indexJSON).Decode(&index); err != nil {
```

Comment:
This is shorter than `ReadAll` + `Unmarshal`, but doesn’t it also silently accept garbage (or a valid JSON?) after the processed object? And does that matter? (Quite possibly it doesn’t…)